src/main/java/frc/robot/subsystems/intake/commands/IntakeByVision.java

```diff
@@ -0,0 +1,52 @@
+package frc.robot.subsystems.intake.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.intake.Intake;
+
+import java.util.function.BooleanSupplier;
+
+/**
+ * Default intake but only sucks alliance colored balls.
+ */
+public class IntakeByVision extends CommandBase {

```

Comment:
This command is very similar to `BasicIntakeCommand`. I recommend deleting this command and adding a `BooleanSupplier` named `shouldIntake` or something similar to `BasicIntakeCommand` instead.

Reply:
I agree this would shorten the code